entity-service/src/main/kotlin/com/egm/stellio/entity/service/SubscriptionEventListenerService.kt

```diff
@@ -0,0 +1,60 @@
+package com.egm.stellio.entity.service
+
+import com.egm.stellio.shared.model.*
+import com.egm.stellio.shared.util.JsonLdUtils.parseJsonLdFragment
+import com.egm.stellio.shared.util.JsonUtils.parseNotificationEvent
+import com.egm.stellio.shared.util.JsonUtils.parseSubscriptionEvent
+import com.egm.stellio.shared.util.toUri
+import org.slf4j.LoggerFactory
+import org.springframework.cloud.stream.annotation.EnableBinding
+import org.springframework.cloud.stream.annotation.StreamListener
+import org.springframework.stereotype.Component
+
+@Component
+@EnableBinding(SubscriptionSink::class)
+class SubscriptionEventListenerService(
+    private val subscriptionHandlerService: SubscriptionHandlerService
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @StreamListener("cim.subscription")
+    fun processSubscription(content: String) {
+        when (val subscriptionEvent = parseSubscriptionEvent(content)) {
```

Comment:
can't a subscription event conform to the model of an entity event?

Reply:
Yes and the notification event too.
I can re-use the existing entityEvent, and add an ENTITY_UPDATE operationType.